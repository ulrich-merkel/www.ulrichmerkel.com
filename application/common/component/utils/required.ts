/**
 * Es6 module for creating form input required attributes.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isBoolean } from 'lodash';

type RequiredAttributes = {
    'aria-required': boolean;
    required: boolean;
};

/**
 * Get component properties for required form input elements.
 *
 * @param {boolean} required - Whether the input element is required or not
 * @returns {object|null} The component props for spreading
 */
export function getRequiredAttributes(
    required?: boolean
): RequiredAttributes | null {
    return isBoolean(required) && required
        ? {
              'aria-required': true,
              required: true
          }
        : null;
}
