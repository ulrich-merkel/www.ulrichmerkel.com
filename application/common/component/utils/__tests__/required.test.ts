import { getRequiredAttributes } from '../required';

describe('getRequiredAttributes', function fnDescribe() {
    it('should return the correct attributes', function fnIt() {
        expect(getRequiredAttributes()).toMatchSnapshot();
        expect(getRequiredAttributes(false)).toMatchSnapshot();
        expect(getRequiredAttributes(true)).toMatchSnapshot();
    });
});
