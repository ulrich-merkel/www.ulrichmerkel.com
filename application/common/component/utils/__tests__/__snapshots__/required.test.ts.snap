// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRequiredAttributes should return the correct attributes 1`] = `null`;

exports[`getRequiredAttributes should return the correct attributes 2`] = `null`;

exports[`getRequiredAttributes should return the correct attributes 3`] = `
Object {
  "aria-required": true,
  "required": true,
}
`;
